src/macros.jl

```diff
@@ -2161,3 +2161,312 @@ julia> @by df :a begin
 macro by(x, what, args...)
     esc(by_helper(x, what, args...))
 end
+
+##############################################################################
+##
+## @unique - distinct row selection
+##
+##############################################################################
+
+function unique_helper(x, args...)
+    exprs, outer_flags = create_args_vector(args...)
+    t = (fun_to_vec(ex; no_dest=true, outer_flags=outer_flags) for ex in exprs)
+    tmp = gensym("tmp")
+    quote            
+        $(tmp) = DataFrames.select($x, $(t...); copycols = false)
+        if isempty($(args))
+            $DataFrames.unique($x)
+        else
+            rowidxs = ((!).(DataFrames.nonunique($tmp)))
+            ($x)[rowidxs, :]                
+        end
+    end
+end
+
+"""
+    @unique(d, args...)
+
+Return the first occurence of unique rows in an `AbstractDataFrame` according 
+to given combinations of values in selected columns or their transformation. 
+`args` can be any column selector or transformation accepted by `select`.
+
+The implementation of `@unique` differs from the `unique` function in DataFrames.jl. 
```

Comment:
This paragraph is too technical. 

Please create a `### Details` section beneath `### Returns` to discuss the implementation. The main thing the user needs to know is that 

```
unique(df, [:a, :b])
```

is different from `@unique(df, [:a, :b])`

Reply:
Okay I have the solution. 

Follow the implementation of `@orderby` and use `gensym_names = true`. That will fix the errors.

Reply:
I'm happy to be wrong, but I find that`@orderby(df, [:x, :z, :d])` results in the same error as above on `v0.12.0`, and that following the implementation of `@orderby` doesn't seem to resolve these errors.

Reply:
That's fine. If the user understands that it uses `select` machinery under the hood, and we communicate that `unique(df, [:x, :y, :z])` is fundamentally different than `@unique(df, [:x, :y, :z])` then we are fine. 

The problem is that 

```
@unique(df, [:x, :y, :z])
```

creates a three element array and tries to make that a column. If the data frame is a different length, it will error. 

The major problem here is that `@unique(df, [:x, :y, :z])` *will* work if the DataFrames has 3 rows... 

I think this is okay though. As long as users know what it's doing.

Reply:
I wonder if this deviates enough from `DataFrames.unique` to merit a new name? Similar to `sort` and `@orderby`. Maybe we should call this `@distinct`? I hope we don't get sued by rstudio for that. 

cc @bkamins @nalimilan

Reply:
So the deviation is that you want to allow `@distinct(df, :a, :b, :c)`? While `@distinct(df, $[:a, :b, :c])` is required as `@distinct(df, [:a, :b, :c])` errors? Then I think it would be better to have a new name.

Reply:
Yes. 

I agree, a new name is probably better here. @MatthewRGonzalez can you change to `@distinct`?

Reply:
Sure. I'll commit with some updated documentation later on.

Reply:
Thanks!